### PR TITLE
kparse script

### PR DESCRIPTION
**Pull request by dwightguth**
This PR is marked as draft because this script won't function until #2022 is merged. However, it is ready for review.

### CODE REVIEW

File: k-distribution/src/main/scripts/bin/kparse [hunk 1]
```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env bash
+set -e
+set -o pipefail
+set -u
+
+export PATH="$(cd "$(dirname "$0")"; pwd):$PATH"
+
+# initialize flags
+ARGV=()
```
Comment (ehildenb) [inline k-distribution/src/main/scripts/bin/kparse:9]: Bash associative arrays on MacOS issue.

File: k-distribution/src/main/scripts/bin/kparse [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+print_usage () {
+cat <<HERE
+Usage: $KPARSE <INPUT_FILE> [-d DIR] [-m MODULE] [-s SORT]
+Generate an on-the-fly parser from a K definition and use it to parse a term.
+
+Mandatory arguments to long options are mandatory for short options too.
+
+$KPARSE expects exactly 1 positional argument. It must be either a readable
+file, or '-'. By default, this file will be parsed using the main syntax
```
Comment (ehildenb) [inline k-distribution/src/main/scripts/bin/kparse]:
```suggestion
file, or '-' (for `stdin`). By default, this file will be parsed using the main syntax
```

File: k-distribution/src/main/scripts/bin/kparse [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+      --no-exc-wrap        Do not wrap exception messages to 80 chars. Keep
+                           long lines.
+      --output-file FILE   Print resulting AST to FILE
+      --save-temps         Do not delete temporary files when $KPARSE
+                           terminates.
+  -s, --sort               Sort to use as start symbol of parser
+  -v, --verbose            Print significant commands executed by $KPARSE.
+      --version            Print version of K
+      --                   Interpret all arguments following as positional
+                           arguments.
```
Comment (ehildenb) [inline k-distribution/src/main/scripts/bin/kparse]: Periods on sentences.

File: k-distribution/src/main/scripts/bin/kparse [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+Generate an on-the-fly parser from a K definition and use it to parse a term.
+
+Mandatory arguments to long options are mandatory for short options too.
+
+$KPARSE expects exactly 1 positional argument. It must be either a readable
+file, or '-'. By default, this file will be parsed using the main syntax
+module of the K definition and the sort of the \$PGM configuration variable.
+The resulting AST, in KORE format, will be output on stdout. This can be
+customized; see below. If no \$PGM configuration variable exists, the default
+sort will be the KItem sort.
```
Comment (ehildenb) [inline k-distribution/src/main/scripts/bin/kparse]: This is a bit confusing to me. Are you saying just that the output on stdout can be customized? Then I would say "will be output on stdout (or to a file, as specified, see below).". If you mean that the output format AST can be changed (to KAST or something), then I thought that was a non-goal of this tool?

File: k-distribution/src/main/scripts/bin/kparse (hunk 1 above)
Comment (dwightguth) [reply k-distribution/src/main/scripts/bin/kparse:9]: This isn't an associative array. Just a regular array, which is fine.

File: k-distribution/src/main/scripts/bin/kparse (hunk 3 above)
Comment (dwightguth) [reply k-distribution/src/main/scripts/bin/kparse]: I couldn't tell if you were saying that there should be or should not be periods, but I looked at other --help messages for existing linux tools and they seem to generally not use periods, so I've made it consistent that way.

File: k-distribution/src/main/scripts/bin/kparse [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+      case "$2" in
+        program) ;;
+        kast) ;;
+        binary) ;;
+        json) ;;
+        kore) ;;
+
+        *)
+	error 'Invalid value for --input. Should be one of "program", "kast", "binary", "json", or "kore".'
+        ;;
```
Comment (ehildenb) [inline k-distribution/src/main/scripts/bin/kparse]: Formatting here seems messed up. `error` here should be indented one more level right?

Please audit all three of yoru PRs for uses of tabs/spacse and consistent spacing.

**Review by ehildenb — APPROVED** [review]

**Pull request merged**